libnodegl/node_texture.c

```
@@ -357,7 +357,9 @@ static void handle_buffer_frame(struct ngl_node *node)
     struct buffer_priv *buffer = s->data_src->priv_data;
     const uint8_t *data = buffer->data;
 
-    ngli_texture_upload(s->texture, data, 0);
+    int ret = ngli_texture_upload(s->texture, data, 0);
+    if (ret < 0)
+        LOG(ERROR, "could not upload texture buffer");
```

This is called by `texture_update()` which does return an error code: can we take the opportunity to add a commit that checks and raise the errors from `handle_buffer_frame()` and `handle_media_frame()`?

---

This needs investigations. Do we want to stop rendering of the entire frame if we fail to map a media frame (instead of using a dummy black texture) ?

---

Maybe not; in that case maybe we should print the message as a warning instead to show that's not intended to be a hard fail?

---

Even though we are not stopping rendering, I think we should keep the messages as errors for now as they are quite important, more than just a regular warning depending on the root cause. I'm not decided yet on whether we should stop rendering or not.

---

OK well, I just want to make sure the next person reading this code doesn't think this is an overlooked issue. I felt like a warning was transmitting this idea well, but a comment would work as well (`/* Tolerate texture upload failure for media sources because they're a "likely" errors where we prefer to black-out part of the presentation instead of hard-failing */` or whatever)

---

I've updated the first commit message and added a new one. The first commit makes the code error out if there is an issue while uploading a regular buffer to the texture. The second one makes it more explicit why we ignore media frames mapping/upload failures.